internal/cri/server/container_remove.go

```diff
@@ -124,6 +124,16 @@ func (c *criService) RemoveContainer(ctx context.Context, r *runtime.RemoveConta
 			volatileContainerRootDir, err)
 	}
 
+	ociRuntime, err := c.config.GetSandboxRuntime(sandbox.Config, sandbox.Metadata.RuntimeHandler)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get sandbox runtime: %w", err)
+	}
+
+	err = c.mutateUnmounts(ctx, container.Config.Mounts, c.RuntimeSnapshotter(ctx, ociRuntime))
```

Comment:
If there are multiple containers sharing one image volume, one container exits, and this deletion will impact existing running? Even if we have mount namespace, I think it's better to cleanup mounts during deleting sandbox.

Reply:
agree should wait to remove until the pod is removed

Reply:
will make restarted containers less expensive

Reply:
will need to move the new image volumes to the pod metadata to facilitate..